explainerdashboard/explainer_methods.py

```diff
@@ -1791,3 +1796,25 @@ def get_xgboost_preds_df(xgbmodel, X_row, pos_label=1):
             0, "pred_proba"
         ]
     return xgboost_preds_df
+
+
+def check_dtype_of(
```

Comment:
could you add some tests for this? How flexible is it? (e.g. will it break over float32 vs float64? int vs float? etc)